http/src/config.rs

```diff
@@ -37,8 +59,18 @@ pub enum InitializationError {
 pub fn initialize(
     ipfs_path: &Path,
     bits: NonZeroU16,
-    profiles: Vec<String>,
+    profiles: Vec<Profile>,
 ) -> Result<(), InitializationError> {
+    // This check is done here to avoid an empty config file being created in the case of an
+    // unsupported input.
```

Comment:
```suggestion
```

I think this comment is covered by the `unimplemented!` and the following `profiles[0]` on the https://github.com/rs-ipfs/rust-ipfs/blob/4b58f0cdf77c1c6d5a277da8ccf26f6b8fe9da9f/http/src/config.rs#L93-L96 I don't think it's possible to fallthrough here and get an empty list for addresses as the `profiles[0]` would had panicked.

Given the panic and existing failure possibilities in `create` I am now thinking if the address (and most other parts of CompatibleConfigFile) should be produced by or with the help of the profiles before the file is even created. The existing structure of this simple config file creation does seem quite unfortunate..

Perhaps a more drastic restructuring is in order? A panic or failure from `create` seems to currently leave behind an empty file. A more drastic restructuring would create the in-memory CompatibleConfigFile fully and only serialize it as the last step.

Reply:
Aah apologies, read the original comment wrong, you were right on the empty config file which I misread as empty `.Addresses.API` :)